### ISSUE TITLE
Messages not getting deleted from the cluster nodes after they are consumed when with work queue policy is aggregated on remote stream

### ISSUE THREAD

**Issue by PKuchibhatla** [nats-io/nats-server#3795]
![image](https://user-images.githubusercontent.com/36994968/213680715-20a46d16-ea9d-4f67-952d-e3ff9fe602a0.png)

**Comment by derekcollison** [comment]
We need server versions, topology setup, JetStream asset info (`nats stream info <stream_name>`) etc.

**Comment by PKuchibhatla** [comment]
Stream on Leaf 1.
![image](https://user-images.githubusercontent.com/36994968/213966089-8e69bb1a-86b8-4e97-bc73-8ebca097d14f.png)
Stream on Leaf 2.
![image](https://user-images.githubusercontent.com/36994968/213966151-7ecb23ad-57ee-4733-98bf-2d23583c4e67.png)
Stream on Hub

![image](https://user-images.githubusercontent.com/36994968/213966321-8c030f93-5e08-4cc7-a112-d50f7acfd137.png)
![image](https://user-images.githubusercontent.com/36994968/213966364-1538de63-4bd0-46d5-9719-82328c4b0001.png)

Topology - HUB Cluster - Leaf Clusters. 
![image](https://user-images.githubusercontent.com/36994968/213966454-075d75a1-af5d-465e-b17c-8dcf1f203a8c.png)

Stream Aggregation form Leaf's to Hub
![image](https://user-images.githubusercontent.com/36994968/213966587-42294dd5-adca-4aba-9897-5ffa74f2cf8b.png)

**Comment by derekcollison** [comment]
When messages are sourced or mirrored they are deleted independently. Deletes from the origin are not propagated.

**Comment by PKuchibhatla** [comment]
i think this  will use lot of space and does it seem to be a bug in case of cluster mode to sync the data across nodes.

**Comment by derekcollison** [comment]
This was a deliberate design decision early on to allow mirrors and sources to have their own parameters, such as limits.

So in a limit case, if hub said max age 1hr and the leaf said 2hrs, if we propagated deletes we would be doing the wrong thing on the leafnode.

So the decision was made to not propagate deletes and each stream handles things on its own.

I realize that if you are trying to source or mirror an interest-policy or workqueue this could be an issue. But again, consider the use case where the mirror is for audit and governance, you would not want them being deleted.

What role does the leafnode play here? The message stored there are being used for what purpose?

**Comment by PKuchibhatla** [comment]
i was referring to the workqueue policy which is on msg ack the message should get deleted right.

in leaf iam having some applications which process the msg for delivery to a callback uri

**Comment by derekcollison** [comment]
Yes they should get deleted but could be held up waiting on transfer to the hub.

Apologies I misunderstood, will take a look.

**Comment by derekcollison** [comment]
So their are applications connected to the WQ streams in the leaf clusters, and those streams are being aggregated to the hub stream as well, also in WQ mode.

And when applications on the leafs consume the messages on that stream are not being removed correct?
And the lead connection from the leaf to the hub is active and connected or disconnected?

**Comment by PKuchibhatla** [comment]
And when applications on the leafs consume the messages on that stream are not being removed correct? ==> Yes
And the lead connection from the leaf to the hub is active and connected or disconnected ==> Connected

**Comment by derekcollison** [comment]
Interest based retention and deletion, as well as user initiated deletion, does not propagate across mirrors or sources by design. They will operate independently from one another.

**Comment by PKuchibhatla** [comment]
Is this the same applicable to WorkQ Policy as well 
In case if we want to have this propagation happen how can we achieve this..
